app/src/main/java/com/example/atheneum/activities/BookInfoActivity.java

```diff
@@ -120,10 +127,48 @@ public void onChanged(@Nullable Book book) {
                     textIsbn.setText(String.valueOf(book.getIsbn()));
                     textDesc.setText(book.getDescription());
                     textStatus.setText(String.valueOf(book.getStatus()));
+                    borrowerID = book.getBorrowerID();
+
+                    // using borrowerID, show borrower email and profile image, or "None"
+                    final TextView borrowerEmailTextView = (TextView) findViewById(R.id.book_borrower_email);
+                    final ImageView borrowerPicture = (ImageView) findViewById(R.id.borrower_profile_pic);
+                    // retrieve the User's email
+                    if (!(borrowerID == null || borrowerID.equals(""))) {
+                        Log.i(TAG, "Valid borrowerID");
+
+                        // retrieve email
+                        UserViewModelFactory userViewModelFactory = new UserViewModelFactory(borrowerID);
+                        userViewModel = ViewModelProviders.of(BookInfoActivity.this, userViewModelFactory).get(UserViewModel.class);
+                        final LiveData<User> userLiveData = userViewModel.getUserLiveData();
+
+                        userLiveData.observe(BookInfoActivity.this, new Observer<User>() {
+                            @Override
+                            public void onChanged(@Nullable User user) {
+                                Log.i(TAG, "in Observer!");
+                                // update borrower email
+                                borrowerEmailTextView.setText(user.getUserName());
+                                Log.i(TAG, "User email*** " + user.getUserName());
+
+                                String userPic = user.getPhotos().get(0);
```

Comment:
This causes a crash since not every user actually has profile pictures. You should get the list of photos and then check if it's empty before you do this:

I added your userID as a borrower in one of my books and it crashed since you don't have a profile picture. I also took a [screen recording](https://drive.google.com/open?id=1pDMGFwyPPSKlhJTaKzAbnmE9HVBAmYG_) of this.

```
2019-03-09 13:14:06.970 11049-11049/com.example.atheneum E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.example.atheneum, PID: 11049
    java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
        at java.util.ArrayList.get(ArrayList.java:411)
        at com.example.atheneum.activities.BookInfoActivity$1$1.onChanged(BookInfoActivity.java:152)
        at com.example.atheneum.activities.BookInfoActivity$1$1.onChanged(BookInfoActivity.java:144)
        at android.arch.lifecycle.LiveData.considerNotify(LiveData.java:109)
        at android.arch.lifecycle.LiveData.dispatchingValue(LiveData.java:126)
        at android.arch.lifecycle.LiveData.setValue(LiveData.java:282)
        at android.arch.lifecycle.MutableLiveData.setValue(MutableLiveData.java:33)
        at android.arch.lifecycle.Transformations$1.onChanged(Transformations.java:70)
        at android.arch.lifecycle.MediatorLiveData$Source.onChanged(MediatorLiveData.java:152)
        at android.arch.lifecycle.LiveData.considerNotify(LiveData.java:109)
        at android.arch.lifecycle.LiveData.dispatchingValue(LiveData.java:126)
        at android.arch.lifecycle.LiveData.setValue(LiveData.java:282)
        at com.example.atheneum.utils.FirebaseQueryLiveData.access$400(FirebaseQueryLiveData.java:20)
        at com.example.atheneum.utils.FirebaseQueryLiveData$MyValueEventListener.onDataChange(FirebaseQueryLiveData.java:94)
        at com.google.firebase.database.core.ValueEventRegistration.fireEvent(com.google.firebase:firebase-database@@16.0.6:75)
        at com.google.firebase.database.core.view.DataEvent.fire(com.google.firebase:firebase-database@@16.0.6:63)
        at com.google.firebase.database.core.view.EventRaiser$1.run(com.google.firebase:firebase-database@@16.0.6:55)
        at android.os.Handler.handleCallback(Handler.java:761)
        at android.os.Handler.dispatchMessage(Handler.java:98)
        at android.os.Looper.loop(Looper.java:156)
        at android.app.ActivityThread.main(ActivityThread.java:6523)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:942)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:832)
```

Reply:
Added a check for no pictures. Thanks!